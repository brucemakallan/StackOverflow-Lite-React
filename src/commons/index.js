const URL = 'https://stackoverflow-lite3-abm.herokuapp.com/api/v1';
export default URL;
