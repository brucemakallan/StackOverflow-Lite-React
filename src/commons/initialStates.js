const initialState = {
	signupReducer: {
		user: {
			fullName: '',
			email: '',
			password: '',
			retypePassword: ''
		}
	}
};

export const sampleUser = {
	fullName: 'Jack Katto',
	email: 'jack.katto@gmail.com',
	password: 'password#123',
	retypePassword: 'password#123'
};

export default initialState;
