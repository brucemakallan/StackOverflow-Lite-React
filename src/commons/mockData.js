const registrationMockData = {
	full_name: 'James Katto',
	email: 'jameskatto@gmail.com',
	password: 'password#123',
	retype_password: 'password#123'
};

export const loginMockData = {
	email: 'jameskatto@gmail.com',
	password: 'password#123'
};

export default registrationMockData;
