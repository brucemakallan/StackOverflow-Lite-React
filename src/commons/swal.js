export const swalError = {
	type: 'error',
	title: 'Oops...',
	text: 'Something went wrong!'
};

export const swalSuccess = {
	type: 'success',
	title: 'All good',
	text: 'Operation Successful'
};
