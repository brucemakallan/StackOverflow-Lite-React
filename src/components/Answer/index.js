import React from 'react';
import PropTypes from 'prop-types';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
	faCheck, faPencilAlt, faTrash, faThumbsUp, faThumbsDown
} from '@fortawesome/free-solid-svg-icons';
import '../Question/question.scss';

library.add(faCheck);
library.add(faPencilAlt);
library.add(faTrash);
library.add(faThumbsDown);
library.add(faThumbsUp);

const Answer = ({ answer }) => (
	<div className="answer-card">
		<div className="title">{answer.answer}</div>
		<div className="date small">{answer.date_posted}</div>
		<div className="answer-footer">
			<div className="votes children">
				<FontAwesomeIcon className="icon" icon="thumbs-up" title="Up-vote" />
				<FontAwesomeIcon className="icon" icon="thumbs-down" title="Down-vote" />
				<span className="accent-color">{answer.votes} Votes</span>
			</div>
			<div className="options children">
				<FontAwesomeIcon className="icon green" icon="check" title="Accept" />
				<FontAwesomeIcon className="icon green" icon="pencil-alt" title="Edit" />
				<FontAwesomeIcon className="icon red" icon="trash" title="Delete" />
			</div>
		</div>
	</div>
);

Answer.propTypes = {
	answer: PropTypes.shape({}).isRequired
};

export default Answer;
