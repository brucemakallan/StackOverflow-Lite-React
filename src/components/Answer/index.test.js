import React from 'react';
import { shallow } from 'enzyme';
import Answer from '.';
import { sampleAnswer } from '../../commons/initialStates';

describe('<Answer /> component tests', () => {
	test('renders correctly', () => {
		const wrapper = shallow(<Answer answer={sampleAnswer} />);
		expect(wrapper).toMatchSnapshot();
	});
});
