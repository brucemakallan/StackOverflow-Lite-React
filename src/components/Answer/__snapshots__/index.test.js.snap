// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Answer /> component tests renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Answer
    answer={
      Object {
        "accepted": false,
        "answer": "This is a sample answer",
        "date_posted": "2018-12-28 07:43:31",
        "id": 2,
        "question_id": 1,
        "user_id": 13,
        "votes": 12,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
        >
          This is a sample answer
        </div>,
        <div
          className="date small"
        >
          2018-12-28 07:43:31
        </div>,
        <div
          className="answer-footer"
        >
          <div
            className="votes children"
          >
            <FontAwesomeIcon
              border={false}
              className="icon"
              fixedWidth={false}
              flip={null}
              icon="thumbs-up"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Up-vote"
              transform={null}
            />
            <FontAwesomeIcon
              border={false}
              className="icon"
              fixedWidth={false}
              flip={null}
              icon="thumbs-down"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Down-vote"
              transform={null}
            />
            <span
              className="accent-color"
            >
              12
               Votes
            </span>
          </div>
          <div
            className="options children"
          >
            <FontAwesomeIcon
              border={false}
              className="icon green"
              fixedWidth={false}
              flip={null}
              icon="check"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Accept"
              transform={null}
            />
            <FontAwesomeIcon
              border={false}
              className="icon green"
              fixedWidth={false}
              flip={null}
              icon="pencil-alt"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Edit"
              transform={null}
            />
            <FontAwesomeIcon
              border={false}
              className="icon red"
              fixedWidth={false}
              flip={null}
              icon="trash"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Delete"
              transform={null}
            />
          </div>
        </div>,
      ],
      "className": "answer-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is a sample answer",
          "className": "title",
        },
        "ref": null,
        "rendered": "This is a sample answer",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2018-12-28 07:43:31",
          "className": "date small",
        },
        "ref": null,
        "rendered": "2018-12-28 07:43:31",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="votes children"
            >
              <FontAwesomeIcon
                border={false}
                className="icon"
                fixedWidth={false}
                flip={null}
                icon="thumbs-up"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Up-vote"
                transform={null}
              />
              <FontAwesomeIcon
                border={false}
                className="icon"
                fixedWidth={false}
                flip={null}
                icon="thumbs-down"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Down-vote"
                transform={null}
              />
              <span
                className="accent-color"
              >
                12
                 Votes
              </span>
            </div>,
            <div
              className="options children"
            >
              <FontAwesomeIcon
                border={false}
                className="icon green"
                fixedWidth={false}
                flip={null}
                icon="check"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Accept"
                transform={null}
              />
              <FontAwesomeIcon
                border={false}
                className="icon green"
                fixedWidth={false}
                flip={null}
                icon="pencil-alt"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Edit"
                transform={null}
              />
              <FontAwesomeIcon
                border={false}
                className="icon red"
                fixedWidth={false}
                flip={null}
                icon="trash"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Delete"
                transform={null}
              />
            </div>,
          ],
          "className": "answer-footer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FontAwesomeIcon
                  border={false}
                  className="icon"
                  fixedWidth={false}
                  flip={null}
                  icon="thumbs-up"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title="Up-vote"
                  transform={null}
                />,
                <FontAwesomeIcon
                  border={false}
                  className="icon"
                  fixedWidth={false}
                  flip={null}
                  icon="thumbs-down"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title="Down-vote"
                  transform={null}
                />,
                <span
                  className="accent-color"
                >
                  12
                   Votes
                </span>,
              ],
              "className": "votes children",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "icon",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "thumbs-up",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "Up-vote",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "icon",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "thumbs-down",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "Down-vote",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    12,
                    " Votes",
                  ],
                  "className": "accent-color",
                },
                "ref": null,
                "rendered": Array [
                  12,
                  " Votes",
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FontAwesomeIcon
                  border={false}
                  className="icon green"
                  fixedWidth={false}
                  flip={null}
                  icon="check"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title="Accept"
                  transform={null}
                />,
                <FontAwesomeIcon
                  border={false}
                  className="icon green"
                  fixedWidth={false}
                  flip={null}
                  icon="pencil-alt"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title="Edit"
                  transform={null}
                />,
                <FontAwesomeIcon
                  border={false}
                  className="icon red"
                  fixedWidth={false}
                  flip={null}
                  icon="trash"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title="Delete"
                  transform={null}
                />,
              ],
              "className": "options children",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "icon green",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "check",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "Accept",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "icon green",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "pencil-alt",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "Edit",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "icon red",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "trash",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "Delete",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
          >
            This is a sample answer
          </div>,
          <div
            className="date small"
          >
            2018-12-28 07:43:31
          </div>,
          <div
            className="answer-footer"
          >
            <div
              className="votes children"
            >
              <FontAwesomeIcon
                border={false}
                className="icon"
                fixedWidth={false}
                flip={null}
                icon="thumbs-up"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Up-vote"
                transform={null}
              />
              <FontAwesomeIcon
                border={false}
                className="icon"
                fixedWidth={false}
                flip={null}
                icon="thumbs-down"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Down-vote"
                transform={null}
              />
              <span
                className="accent-color"
              >
                12
                 Votes
              </span>
            </div>
            <div
              className="options children"
            >
              <FontAwesomeIcon
                border={false}
                className="icon green"
                fixedWidth={false}
                flip={null}
                icon="check"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Accept"
                transform={null}
              />
              <FontAwesomeIcon
                border={false}
                className="icon green"
                fixedWidth={false}
                flip={null}
                icon="pencil-alt"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Edit"
                transform={null}
              />
              <FontAwesomeIcon
                border={false}
                className="icon red"
                fixedWidth={false}
                flip={null}
                icon="trash"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Delete"
                transform={null}
              />
            </div>
          </div>,
        ],
        "className": "answer-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is a sample answer",
            "className": "title",
          },
          "ref": null,
          "rendered": "This is a sample answer",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018-12-28 07:43:31",
            "className": "date small",
          },
          "ref": null,
          "rendered": "2018-12-28 07:43:31",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="votes children"
              >
                <FontAwesomeIcon
                  border={false}
                  className="icon"
                  fixedWidth={false}
                  flip={null}
                  icon="thumbs-up"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title="Up-vote"
                  transform={null}
                />
                <FontAwesomeIcon
                  border={false}
                  className="icon"
                  fixedWidth={false}
                  flip={null}
                  icon="thumbs-down"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title="Down-vote"
                  transform={null}
                />
                <span
                  className="accent-color"
                >
                  12
                   Votes
                </span>
              </div>,
              <div
                className="options children"
              >
                <FontAwesomeIcon
                  border={false}
                  className="icon green"
                  fixedWidth={false}
                  flip={null}
                  icon="check"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title="Accept"
                  transform={null}
                />
                <FontAwesomeIcon
                  border={false}
                  className="icon green"
                  fixedWidth={false}
                  flip={null}
                  icon="pencil-alt"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title="Edit"
                  transform={null}
                />
                <FontAwesomeIcon
                  border={false}
                  className="icon red"
                  fixedWidth={false}
                  flip={null}
                  icon="trash"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title="Delete"
                  transform={null}
                />
              </div>,
            ],
            "className": "answer-footer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FontAwesomeIcon
                    border={false}
                    className="icon"
                    fixedWidth={false}
                    flip={null}
                    icon="thumbs-up"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title="Up-vote"
                    transform={null}
                  />,
                  <FontAwesomeIcon
                    border={false}
                    className="icon"
                    fixedWidth={false}
                    flip={null}
                    icon="thumbs-down"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title="Down-vote"
                    transform={null}
                  />,
                  <span
                    className="accent-color"
                  >
                    12
                     Votes
                  </span>,
                ],
                "className": "votes children",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "icon",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": "thumbs-up",
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "title": "Up-vote",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "icon",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": "thumbs-down",
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "title": "Down-vote",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      12,
                      " Votes",
                    ],
                    "className": "accent-color",
                  },
                  "ref": null,
                  "rendered": Array [
                    12,
                    " Votes",
                  ],
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FontAwesomeIcon
                    border={false}
                    className="icon green"
                    fixedWidth={false}
                    flip={null}
                    icon="check"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title="Accept"
                    transform={null}
                  />,
                  <FontAwesomeIcon
                    border={false}
                    className="icon green"
                    fixedWidth={false}
                    flip={null}
                    icon="pencil-alt"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title="Edit"
                    transform={null}
                  />,
                  <FontAwesomeIcon
                    border={false}
                    className="icon red"
                    fixedWidth={false}
                    flip={null}
                    icon="trash"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title="Delete"
                    transform={null}
                  />,
                ],
                "className": "options children",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "icon green",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": "check",
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "title": "Accept",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "icon green",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": "pencil-alt",
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "title": "Edit",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "icon red",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": "trash",
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "title": "Delete",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
